curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/iterate-through-an-array-with-a-for-loop.english.md

```diff
@@ -16,7 +16,7 @@ for (var i = 0; i < arr.length; i++) {
 }
 ```
 
-Remember that Arrays have zero-based numbering, which means the last index of the array is length - 1. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at length - 1.
+Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops when <code>i</code> is at <code>length - 1</code>.
```

Comment:
@organicdude I think we can make this better by changing a bit more of the wording.  I will let some other mods add their comments before you make the commit though.  What do you think?
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>.
```

Reply:
Sounds good - just please let me know if I've fallen asleep at the wheel and am not allowing a finished commit to go through.

Here's another way to phrase it - more noob friendly IMHO.
```suggestion
Remember that arrays first index is <code>0</code>, not <code>1</code>. That means that if <code>i</code> is your array's index, then the last the last defined index will be <code>length - 1</code>. That also means that your loop should stop running when <code>i > length - 1</code>.
```

Reply:
We'll let you know.  @Manish-Giri @thecodingaviator @ojeytonwilliams @moT01 What do you think about my suggestion?

Reply:
@RandellDawson How does this sound.
```suggestion
Remember that arrays have zero-based indexing, which means the last index of the array is <code>length - 1</code>. Our <dfn>condition</dfn> for this loop is <code>i < arr.length</code>, which stops the loop when <code>i</code> is equal to <code>length</code>, i.e. the last iteration is when <code>i</code> is equal to <code>length-1</code>.
```

Reply:
@organicdude I am leaning toward's @thecodingaviator's version.  That being said, I think just adding on that last part of @thecodingaviator's suggestion makes the last sentence just too long.  I think the example part should be a separate sentence on its own.

Reply:
@RandellDawson Feel free to edit my suggestion or make a new one in this thread to show us what you think. I'm not sure how we would go about splitting it